BitList.cs

```diff
@@ -0,0 +1,1553 @@
+    public void AddFirst(bool bit)
+    {
+        Insert(0, bit);
+    }
+
+    /// <summary>
+    /// Remove a bit in the specified position.
+    /// </summary>
+    /// <exception cref="ArgumentOutOfRangeException" />
+    public void RemoveAt(int index)
+    {
+        if (!IsEmpty && index < Count && index >= 0)
+        {
+            var bools = ToBooleanArray().ToList();
+            bools.RemoveAt(index);
+            bits = new BitArray(bools.ToArray());
+        }
+        else
+            throw new ArgumentOutOfRangeException();
+    }
+
+    /// <summary>
+    /// Remove a range of bits from this instance.
+    /// </summary>
+    /// <param name="index">Start index.</param>
+    /// <param name="count">Count of bits.</param>
+    /// <exception cref="ArgumentOutOfRangeException" />
+    public void RemoveRange(int index, int count)
+    {
+        if (!IsEmpty && index < bits.Length && index >= 0 && count <= bits.Length - index && count > 0)
+        {
+            var bools = ToBooleanArray().ToList();
+            bools.RemoveRange(index, count);
+            bits = new BitArray(bools.ToArray());
+        }
+        else
+            throw new ArgumentOutOfRangeException();
+    }
+
+    /// <summary>
+    /// Remove the specified bit from this instance.
+    /// </summary>
+    public bool Remove(bool bit)
+    {
+        try
+        {
+            RemoveAt(IndexOf(bit));
+            return true;
+        }
+        catch
+        {
+            return false;
+        }
+    }
+
+    /// <summary>
+    /// Remove the specified list of bit from this instance.
+    /// </summary>
+    /// <exception cref="EmptyInstanceException" />
+    public bool Remove(BitList bits)
+    {
+        if (IsEmpty)
+            throw new EmptyInstanceException();
+        else
+        {
+            var thisBitList = ToText();
+            var thoseBitList = bits.ToText();
+            if (!thisBitList.Contains(thoseBitList))
+                return false;
+            var newBitList = thisBitList.Replace(thoseBitList, string.Empty);
+            this.bits = Parse(newBitList).bits;
+            return true;
+        }
+    }
+
+    /// <summary>
+    /// Insert the specified list of bit in the specified index.
+    /// </summary>
+    /// <exception cref="IndexOutOfRangeException" />
+    public void Insert(int index, BitList bits)
+    {
+        if (IsEmpty)
+            this.bits = new BitArray(bits.bits);
+        else if (index < bits.Count && index >= 0)
+        {
+            var bools = ToBooleanArray().ToList();
+            bools.InsertRange(index, bits.ToBooleanArray());
+            this.bits = new BitArray(bools.ToArray());
+        }
+        else
+            throw new IndexOutOfRangeException();
+    }
+
+    /// <summary>
+    /// Insert the specified bit in the specified index.
+    /// </summary>
+    /// <exception cref="IndexOutOfRangeException" />
+    public void Insert(int index, bool bit)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename parameter 'bit' to 'item' to match the interface declaration.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5545753)